lms/djangoapps/learner_home/views.py

```diff
@@ -0,0 +1,183 @@
+    """Get enrollments and enrollment course modes for user"""
+
+    course_enrollments = list(
+        get_course_enrollments(user, org_allow_list, org_block_list, course_limit)
+    )
+
+    # Sort the enrollment pairs by the enrollment date
+    course_enrollments.sort(key=lambda x: x.created, reverse=True)
+
+    # Record how many courses there are so that we can get a better
+    # understanding of usage patterns on prod.
+    monitoring_utils.accumulate("num_courses", len(course_enrollments))
+
+    # Retrieve the course modes for each course
+    enrolled_course_ids = [enrollment.course_id for enrollment in course_enrollments]
+    __, unexpired_course_modes = CourseMode.all_and_unexpired_modes_for_courses(
+        enrolled_course_ids
+    )
+    course_modes_by_course = {
+        course_id: {mode.slug: mode for mode in modes}
+        for course_id, modes in unexpired_course_modes.items()
+    }
+
+    # Construct a dictionary of course mode information
+    # used to render the course list.  We re-use the course modes dict
+    # we loaded earlier to avoid hitting the database.
+    course_mode_info = {
+        enrollment.course_id: complete_course_mode_info(
+            enrollment.course_id,
+            enrollment,
+            modes=course_modes_by_course[enrollment.course_id],
+        )
+        for enrollment in course_enrollments
+    }
+
+    return course_enrollments, course_mode_info
+
+
+def get_email_settings_info(user, course_enrollments):
+    """
+    Given a user and enrollments, determine which courses allow bulk email (show_email_settings_for)
+    and which the learner has opted out from (optouts)
+    """
+    course_optouts = Optout.objects.filter(user=user).values_list(
+        "course_id", flat=True
+    )
+
+    # only show email settings for Mongo course and when bulk email is turned on
+    show_email_settings_for = frozenset(
+        enrollment.course_id
+        for enrollment in course_enrollments
+        if (is_bulk_email_feature_enabled(enrollment.course_id))
+    )
+
+    return show_email_settings_for, course_optouts
+
+
+def get_ecommerce_payment_page(user):
+    """Determine the ecommerce payment page URL if enabled for this user"""
+    ecommerce_service = EcommerceService()
+    return (
+        ecommerce_service.payment_page_url()
+        if ecommerce_service.is_enabled(user)
+        else None
+    )
+
+
+@login_required
+@require_GET
+def dashboard_view(request):  # pylint: disable=unused-argument
+    """List of courses a user is enrolled in or entitled to"""
+
+    # Get user, determine if user needs to confirm email account
+    user = request.user
+    email_confirmation = get_user_account_confirmation_info(user)
+
+    # Determine number of courses to show on dashboard
+    disable_course_limit = request and "course_limit" in request.GET
+    course_limit = get_dashboard_course_limit() if not disable_course_limit else None
```

Comment:
Is this something we're actually using? if this is non-none, do we have a mechanism for backend pagination for the rest of the enrollments?

Reply:
See note in https://2u-internal.atlassian.net/wiki/spaces/PT/pages/15438335/Existing+API+data+flow+discovery#Additional-notes.

**TL;DR** - We are using it, when it overflows there's an option to show all but there is no pagination option. Talking w/ Ben, the decision is to, eventually, remove entirely.